<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
<title>Class system_error</title>
<link rel="stylesheet" href="../../boostbook.css" type="text/css">
<meta name="generator" content="DocBook XSL Stylesheets V1.70.1">
<link rel="start" href="../../index.html" title="Boost.Process">
<link rel="up" href="../../boost_process/reference.html#header.boost.process.exceptions.hpp" title="7.7. Header &lt;boost/process/exceptions.hpp&gt;">
<link rel="prev" href="../../BOOST_PROCESS_POSIX_API.html" title="Macro BOOST_PROCESS_POSIX_API">
<link rel="next" href="not_found_error.html" title="Class template not_found_error">
</head>
<body bgcolor="white" text="black" link="#0000FF" vlink="#840084" alink="#0000FF">
<table cellpadding="2" width="100%"><td valign="top"><img alt="Boost C++ Libraries" width="277" height="86" src="../.././boost.png"></td></table>
<hr>
<div class="spirit-nav">
<a accesskey="p" href="../../BOOST_PROCESS_POSIX_API.html"><img src="../../images/prev.png" alt="Prev"></a><a accesskey="u" href="../../boost_process/reference.html#header.boost.process.exceptions.hpp"><img src="../../images/up.png" alt="Up"></a><a accesskey="h" href="../../index.html"><img src="../../images/home.png" alt="Home"></a><a accesskey="n" href="not_found_error.html"><img src="../../images/next.png" alt="Next"></a>
</div>
<div class="refentry" lang="en">
<a name="boost.process.system_error"></a><div class="titlepage"></div>
<div class="refnamediv">
<h2><span class="refentrytitle">Class system_error</span></h2>
<p>boost::process::system_error &#8212; Run time error initiated in a system call. </p>
</div>
<h2 xmlns:rev="http://www.cs.rpi.edu/~gregod/boost/tools/doc/revision" class="refsynopsisdiv-title">Synopsis</h2>
<div xmlns:rev="http://www.cs.rpi.edu/~gregod/boost/tools/doc/revision" class="refsynopsisdiv"><pre class="synopsis">
<span class="bold"><strong>class</strong></span> system_error {
<span class="bold"><strong>public</strong></span>:
  <span class="emphasis"><em>// types</em></span>
  <span class="bold"><strong>typedef</strong></span> NativeErrorCodeType code_type;  <span class="emphasis"><em>// Native type used by the system to report error codes. </em></span>

  <span class="emphasis"><em>// <a href="system_error.html#boost.process.system_errorconstruct-copy-destruct">construct/copy/destruct</a></em></span>
  <a href="system_error.html#id2379118-bb">system_error</a>(<span class="bold"><strong>const</strong></span> std::string &amp;, <span class="bold"><strong>const</strong></span> std::string &amp;, code_type);
  <a href="system_error.html#id2379154-bb">~system_error</a>(<span class="bold"><strong>void</strong></span>);

  <span class="emphasis"><em>// <a href="system_error.html#id2379073-bb">public member functions</a></em></span>
  <span class="type">code_type</span> <a href="system_error.html#id2379076-bb">code</a>(<span class="bold"><strong>void</strong></span>) <span class="bold"><strong>const</strong></span>;
  <span class="type"><span class="bold"><strong>const</strong></span> <span class="bold"><strong>char</strong></span> *</span> <a href="system_error.html#id2379094-bb">what</a>(<span class="bold"><strong>void</strong></span>) <span class="bold"><strong>const</strong></span>;
};</pre></div>
<div class="refsect1" lang="en">
<a name="id2713237"></a><h2>Description</h2>
<p>The system_error exception wraps the information generated by an operating system error. These errors are the result of incorrect or erroneous system calls. </p>
<div class="refsect2" lang="en">
<a name="id2713242"></a><h3>
<a name="boost.process.system_errorconstruct-copy-destruct"></a><code class="computeroutput">system_error</code> construct/copy/destruct</h3>
<div class="orderedlist"><ol type="1">
<li>
<pre class="literallayout"><a name="id2379118-bb"></a>system_error(<span class="bold"><strong>const</strong></span> std::string &amp; who, <span class="bold"><strong>const</strong></span> std::string &amp; message, 
             code_type sys_err);</pre>
<p>Constructs a new exception based on the result of an incorrect or an erroneous system call.</p>
<p>
</p>
<div class="variablelist"><table border="0">
<col align="left" valign="top">
<tbody><tr>
<td>
<span class="term">Parameters:</span></td>
<td><div class="variablelist"><table border="0">
<col align="left" valign="top">
<tbody>
<tr>
<td>
<span class="term"><code class="computeroutput">message</code></span></td>
<td><p>String that describes the error that occurred (typically the system call that failed). </p></td>
</tr>
<tr>
<td>
<span class="term"><code class="computeroutput">sys_err</code></span></td>
<td><p>The system error code that identifies the error. On a POSIX system this is the value of errno and on a Win32 system it is the value returned by GetLastError(). </p></td>
</tr>
<tr>
<td>
<span class="term"><code class="computeroutput">who</code></span></td>
<td><p>String that indicates the class and function where the error was first detected. </p></td>
</tr>
</tbody>
</table></div></td>
</tr></tbody>
</table></div>
</li>
<li>
<pre class="literallayout"><a name="id2379154-bb"></a>~system_error(<span class="bold"><strong>void</strong></span> );</pre>
<p>Virtual destructor. Does nothing special but is required because of the parent's class destructor. </p>
</li>
</ol></div>
</div>
<div class="refsect2" lang="en">
<a name="id2713418"></a><h3>
<a name="id2379073-bb"></a><code class="computeroutput">system_error</code> public member functions</h3>
<div class="orderedlist"><ol type="1">
<li>
<pre class="literallayout"><span class="type">code_type</span> <a name="id2379076-bb"></a>code(<span class="bold"><strong>void</strong></span> ) <span class="bold"><strong>const</strong></span>;</pre>
<p>Returns the exception's native error code; that is, the value provided by the operating system when it reported the error. </p>
</li>
<li>
<pre class="literallayout"><span class="type"><span class="bold"><strong>const</strong></span> <span class="bold"><strong>char</strong></span> *</span> <a name="id2379094-bb"></a>what(<span class="bold"><strong>void</strong></span> ) <span class="bold"><strong>const</strong></span>;</pre>
<p>Generates and returns a formatted error message for the native error code attached to this exception. This includes the function that raised the error, the system call where it was initiated and the reason for the failure. The last is determined by strerror() in POSIX systems and FormatMessage under Win32.</p>
<p>
</p>
<div class="variablelist"><table border="0">
<col align="left" valign="top">
<tbody><tr>
<td>
<span class="term">Returns:</span></td>
<td><p>A constant C string containing the formatted error message. The caller must not release it. </p></td>
</tr></tbody>
</table></div>
</li>
</ol></div>
</div>
</div>
</div>
<table width="100%"><tr>
<td align="left"></td>
<td align="right"><small>Copyright © 2006 Julio M. Merino Vidal</small></td>
</tr></table>
<hr>
<div class="spirit-nav">
<a accesskey="p" href="../../BOOST_PROCESS_POSIX_API.html"><img src="../../images/prev.png" alt="Prev"></a><a accesskey="u" href="../../boost_process/reference.html#header.boost.process.exceptions.hpp"><img src="../../images/up.png" alt="Up"></a><a accesskey="h" href="../../index.html"><img src="../../images/home.png" alt="Home"></a><a accesskey="n" href="not_found_error.html"><img src="../../images/next.png" alt="Next"></a>
</div>
</body>
</html>
